lib/firestore.js

```diff
@@ -125,12 +124,28 @@ export class FirestoreBackup {
 
   documentRequestLimit: number;
 
+  _PromiseLibrary: Object;
+
   constructor(options: BackupOptions) {
     this.options = Object.assign({}, defaultBackupOptions, options)
 
     if (this.options.requestCountLimit > 1) {
       this.documentRequestLimit = 3 // 3 is the max before diminishing returns
     }
+
+    this.Promise = Promise
+  }
+
+  get Promise() {
+    return this._PromiseLibrary
+  }
+
+  /**
+   * The Promise library used to process asynchronous functions.
+   */
+  set Promise(value: Object) {
+    // TODO: validate this ia valid Promise-like object
+    this._PromiseLibrary = addMapFunction(value)
```

Comment:
Why are we mapping through the promise serial resolver again vs doing them all async? Does this tie into your analysis work?

Reply:
It was a path I thought I wanted to go down until I decided to just write a func to handle promises at a given concurrency. See the latest commits.